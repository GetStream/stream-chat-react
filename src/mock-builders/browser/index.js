export * from './AnalyserNode';
export * from './AudioContext';
export * from './EventEmitter';
export * from './MediaRecorder';
export * from './ResizeObserver';
