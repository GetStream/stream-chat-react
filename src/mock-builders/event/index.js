export { default as dispatchChannelDeletedEvent } from './channelDeleted';
export { default as dispatchChannelHiddenEvent } from './channelHidden';
export { default as dispatchChannelTruncatedEvent } from './channelTruncated';
export { default as dispatchChannelUpdatedEvent } from './channelUpdated';
export { default as dispatchChannelVisibleEvent } from './channelVisible';
export { default as dispatchConnectionChangedEvent } from './connectionChanged';
export { default as dispatchConnectionRecoveredEvent } from './connectionRecovered';
export { default as dispatchMessageDeletedEvent } from './messageDeleted';
export { default as dispatchMessageNewEvent } from './messageNew';
export { default as dispatchMessageUpdatedEvent } from './messageUpdated';
export { default as dispatchNotificationAddedToChannelEvent } from './notificationAddedToChannel';
export { default as dispatchNotificationMessageNewEvent } from './notificationMessageNew';
export { default as dispatchNotificationMutesUpdated } from './notificationMutesUpdated';
export { default as dispatchNotificationRemovedFromChannel } from './notificationRemovedFromChannel';
export { default as dispatchUserUpdatedEvent } from './userUpdated';
