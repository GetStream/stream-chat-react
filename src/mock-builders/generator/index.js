export * from './attachment';
export * from './channel';
export * from './emoji';
export * from './member';
export * from './message';
export * from './reaction';
export * from './user';
export * from './upload';
