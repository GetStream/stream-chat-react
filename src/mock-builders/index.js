/* eslint-disable no-param-reassign */
/* eslint-disable no-underscore-dangle */
import { StreamChat } from 'stream-chat';
import { nanoid } from 'nanoid';

const apiKey = 'API_KEY';
const token = 'dummy_token';

const connectUser = (client, user) =>
  new Promise((resolve) => {
    client.connectionId = 'dumm_connection_id';
    client.user = user;
    client.user.mutes = [];
    client._user = { ...user };
    client.userID = user.id;
    client.userToken = token;
    client.wsPromise = Promise.resolve(true);
    resolve();
  });

function mockClient(client) {
  jest.spyOn(client, '_setToken').mockImplementation();
  jest.spyOn(client, '_setupConnection').mockImplementation();
  jest.spyOn(client, '_setupConnection').mockImplementation();
  jest.spyOn(client, 'getAppSettings').mockImplementation();
  client.tokenManager = {
    getToken: jest.fn(() => token),
    tokenReady: jest.fn(() => true),
  };
  client.connectUser = connectUser.bind(null, client);
  return client;
}

export const getTestClient = () => mockClient(new StreamChat(apiKey));

export const getTestClientWithUser = async (user = { id: nanoid() }) => {
  const client = mockClient(new StreamChat(apiKey));
  await connectUser(client, user);
  return client;
};

export const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; // The maximum is exclusive and the minimum is inclusive
};

export * from './api';
export * from './event';
export * from './generator';
export * from './translator';
export * from './utils';
