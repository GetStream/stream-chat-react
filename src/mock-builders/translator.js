export const mockTranslatorFunction = (value, params = {}) =>
  Object.keys(params).reduce((acc, key) => {
    const regexp = new RegExp(`\\{\\{\\s${key}\\s\\}\\}`, 'g');
    return acc.replace(regexp, params[key]);
  }, value);
