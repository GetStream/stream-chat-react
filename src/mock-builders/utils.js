import { generateChannel, generateMember, generateMessage, generateUser } from './generator';
import { getOrCreateChannelApi, getTestClientWithUser, useMockedApis } from './index';

export async function createClientWithChannel(
  { channelData, empty, existingClient, existingUsers, memberCount, messageCount } = {
    channelData: { image: 'image-xxx', name: 'channel-xxx' },
    empty: false,
    existingClient: null,
    existingUsers: null,
    memberCount: 2,
    messageCount: 20,
  },
) {
  const users = existingUsers || Array.from({ length: memberCount }, generateUser);
  const members = users.map((user) => generateMember({ user }));
  const mockedChannel = generateChannel({
    data: channelData,
    members,
    messages: empty
      ? []
      : ' '
          .repeat(messageCount)
          .split(' ')
          .map((v, i) => generateMessage({ user: users[i % memberCount] })),
  });

  const client = existingClient || (await getTestClientWithUser({ id: users[0].id }));
  useMockedApis(client, [getOrCreateChannelApi(mockedChannel)]); // eslint-disable-line react-hooks/rules-of-hooks
  const channel = client.channel('messaging', mockedChannel.id);
  await channel.watch();

  return { channel, client, users };
}
