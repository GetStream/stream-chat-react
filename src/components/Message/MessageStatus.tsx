import React, { useState } from 'react';
import clsx from 'clsx';

import { DeliveredCheckIcon, MessageDeliveredIcon } from './icons';
import { getReadByTooltipText, mapToUserNameOrId, TooltipUsernameMapper } from './utils';

import { AvatarProps, Avatar as DefaultAvatar } from '../Avatar';
import { LoadingIndicator } from '../Loading';
import { PopperTooltip, Tooltip } from '../Tooltip';
import { useEnterLeaveHandlers } from '../Tooltip/hooks';

import { useChatContext } from '../../context/ChatContext';
import { useComponentContext } from '../../context/ComponentContext';
import { useMessageContext } from '../../context/MessageContext';
import { useTranslationContext } from '../../context/TranslationContext';

import type { DefaultStreamChatGenerics } from '../../types/types';

export type MessageStatusProps = {
  /* Custom UI component to display a user's avatar (overrides the value from `ComponentContext`) */
  Avatar?: React.ComponentType<AvatarProps>;
  /* Message type string to be added to CSS class names. */
  messageType?: string;
  /* Allows to customize the username(s) that appear on the message status tooltip */
  tooltipUserNameMapper?: TooltipUsernameMapper;
};

const UnMemoizedMessageStatus = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
>(
  props: MessageStatusProps,
) => {
  const {
    Avatar: propAvatar,
    messageType = 'simple',
    tooltipUserNameMapper = mapToUserNameOrId,
  } = props;

  const { handleEnter, handleLeave, tooltipVisible } = useEnterLeaveHandlers<HTMLSpanElement>();

  const { client } = useChatContext<StreamChatGenerics>('MessageStatus');
  const { Avatar: contextAvatar } = useComponentContext<StreamChatGenerics>('MessageStatus');
  const {
    isMyMessage,
    lastReceivedId,
    message,
    readBy,
    threadList,
  } = useMessageContext<StreamChatGenerics>('MessageStatus');
  const { t } = useTranslationContext('MessageStatus');
  const { themeVersion } = useChatContext('MessageStatus');
  const [referenceElement, setReferenceElement] = useState<HTMLSpanElement | null>(null);

  const Avatar = propAvatar || contextAvatar || DefaultAvatar;

  if (!isMyMessage() || message.type === 'error') return null;

  const justReadByMe = readBy?.length === 1 && readBy[0].id === client.user?.id;
  const rootClassName = `str-chat__message-${messageType}-status str-chat__message-status`;

  const sending = message.status === 'sending';
  const delivered = message.status === 'received' && message.id === lastReceivedId && !threadList;
  const deliveredAndRead = !!(readBy?.length && !threadList && !justReadByMe);

  const [lastReadUser] = deliveredAndRead
    ? readBy.filter((item) => item.id !== client.user?.id)
    : [];

  return (
    <span
      className={rootClassName}
      data-testid={clsx({
        'message-status-read-by': deliveredAndRead,
        'message-status-received': delivered && !deliveredAndRead,
        'message-status-sending': sending,
      })}
      onMouseEnter={handleEnter}
      onMouseLeave={handleLeave}
      ref={setReferenceElement}
    >
      {sending && (
        <>
          {themeVersion === '1' && <Tooltip>{t<string>('Sending...')}</Tooltip>}
          {themeVersion === '2' && (
            <PopperTooltip
              offset={[0, 5]}
              referenceElement={referenceElement}
              visible={tooltipVisible}
            >
              {t<string>('Sending...')}
            </PopperTooltip>
          )}
          <LoadingIndicator />
        </>
      )}

      {delivered && !deliveredAndRead && (
        <>
          {themeVersion === '1' && <Tooltip>{t<string>('Delivered')}</Tooltip>}
          {themeVersion === '2' && (
            <PopperTooltip
              offset={[0, 5]}
              referenceElement={referenceElement}
              visible={tooltipVisible}
            >
              {t<string>('Delivered')}
            </PopperTooltip>
          )}
          {themeVersion === '2' ? <MessageDeliveredIcon /> : <DeliveredCheckIcon />}
        </>
      )}

      {deliveredAndRead && (
        <>
          {themeVersion === '1' && (
            <Tooltip>{getReadByTooltipText(readBy, t, client, tooltipUserNameMapper)}</Tooltip>
          )}
          {themeVersion === '2' && (
            <PopperTooltip
              offset={[0, 5]}
              referenceElement={referenceElement}
              visible={tooltipVisible}
            >
              {getReadByTooltipText(readBy, t, client, tooltipUserNameMapper)}
            </PopperTooltip>
          )}
          <Avatar
            image={lastReadUser.image}
            name={lastReadUser.name || lastReadUser.id}
            size={15}
            user={lastReadUser}
          />

          {readBy.length > 2 && (
            <span
              className={`str-chat__message-${messageType}-status-number`}
              data-testid='message-status-read-by-many'
            >
              {readBy.length - 1}
            </span>
          )}
        </>
      )}
    </span>
  );
};

export const MessageStatus = React.memo(UnMemoizedMessageStatus) as typeof UnMemoizedMessageStatus;
