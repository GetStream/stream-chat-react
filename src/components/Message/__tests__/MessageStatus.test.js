import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import {
  ChatProvider,
  ComponentProvider,
  MessageProvider,
  TranslationProvider,
} from '../../../context';
import { MessageStatus } from '../MessageStatus';
import { generateMessage, generateUser, getTestClientWithUser } from '../../../mock-builders';

const MESSAGE_STATUS_SENDING_TEST_ID = 'message-status-sending';
const MESSAGE_STATUS_DELIVERED_TEST_ID = 'message-status-received';
const MESSAGE_STATUS_READ_TEST_ID = 'message-status-read-by';
const MESSAGE_STATUS_READ_COUNT_TEST_ID = 'message-status-read-by-many';

const rootClassName = `str-chat__message-simple-status str-chat__message-status`;

const user = { id: 'me' };
const foreignMsg = {
  __html: '<p>regular</p>',
  attachments: [],
  created_at: '2024-05-28T15:13:20.899Z',
  html: '<p>regular</p>',
  id: '5kIE4fIArv11V4YHYdXKO',
  mentioned_users: [],
  pinned_at: null,
  status: 'received',
  text: 'udSNfyk7Z-0MRn17WUQwY',
  type: 'regular',
  updated_at: '2024-05-28T15:13:20.900Z',
  user: null,
};

const ownMessage = generateMessage({ user });
const errorMsg = { ...foreignMsg, type: 'error', user };
const sendingMsg = { ...foreignMsg, status: 'sending', user };
const deliveredMsg = { ...foreignMsg, user };
const readByOthers = [{ id: 'other-user' }];
const t = jest.fn((s) => s);

const defaultMsgCtx = {
  isMyMessage: jest.fn().mockReturnValue(true),
};

const renderComponent = ({ chatCtx, componentCtx, messageCtx, props } = {}) =>
  render(
    <ChatProvider value={chatCtx}>
      <TranslationProvider value={{ t }}>
        <ComponentProvider value={componentCtx || {}}>
          <MessageProvider value={{ ...defaultMsgCtx, ...messageCtx }}>
            <MessageStatus {...props} />
          </MessageProvider>
        </ComponentProvider>
      </TranslationProvider>
    </ChatProvider>,
  );

describe('MessageStatus', () => {
  it('renders nothing for not own messages', async () => {
    const client = await getTestClientWithUser(user);
    const { container } = renderComponent({
      chatCtx: { client },
      messageCtx: { isMyMessage: () => false, message: foreignMsg },
    });
    expect(container).toBeEmptyDOMElement();
  });
  it('renders nothing for error messages', async () => {
    const client = await getTestClientWithUser(user);
    const { container } = renderComponent({
      chatCtx: { client },
      messageCtx: { message: errorMsg },
    });
    expect(container).toBeEmptyDOMElement();
  });

  it('renders default sending UI for the last message', async () => {
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { lastReceivedId: sendingMsg.id, message: sendingMsg },
    });
    expect(screen.getByTestId(MESSAGE_STATUS_SENDING_TEST_ID)).toMatchSnapshot();
  });

  it('renders custom sending UI for the last message', async () => {
    const text = 'CustomMessageSendingStatus';
    const MessageSendingStatus = () => <div>{text}</div>;
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { lastReceivedId: sendingMsg.id, message: sendingMsg },
      props: { MessageSendingStatus },
    });
    expect(screen.getByText(text)).toBeInTheDocument();
  });

  it('renders default sending UI for not the last message', async () => {
    const client = await getTestClientWithUser(user);
    renderComponent({ chatCtx: { client }, messageCtx: { message: sendingMsg } });
    expect(screen.getByTestId(MESSAGE_STATUS_SENDING_TEST_ID)).toMatchSnapshot();
  });

  it('renders custom sending UI for not the last message', async () => {
    const text = 'CustomMessageSendingStatus';
    const MessageSendingStatus = () => <div>{text}</div>;
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { message: sendingMsg },
      props: { MessageSendingStatus },
    });
    expect(screen.getByText(text)).toBeInTheDocument();
  });

  // here
  it('renders default delivered UI for the last message', async () => {
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { lastReceivedId: deliveredMsg.id, message: deliveredMsg },
    });
    expect(screen.getByTestId(MESSAGE_STATUS_DELIVERED_TEST_ID)).toMatchSnapshot();
  });

  it('renders custom delivered UI for the last message', async () => {
    const text = 'CustomMessageDeliveredStatus';
    const MessageDeliveredStatus = () => <div>{text}</div>;
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { lastReceivedId: deliveredMsg.id, message: deliveredMsg },
      props: { MessageDeliveredStatus },
    });
    expect(screen.getByText(text)).toBeInTheDocument();
  });

  it('renders empty container without default delivered UI for not the last message', async () => {
    const client = await getTestClientWithUser(user);
    const { container } = renderComponent({
      chatCtx: { client },
      messageCtx: { message: deliveredMsg },
    });
    expect(container.children[0]).toHaveClass(rootClassName);
    expect(container.children[0]).toBeEmptyDOMElement();
  });

  it('renders empty container without custom delivered UI for not the last message', async () => {
    const text = 'CustomMessageDeliveredStatus';
    const MessageDeliveredStatus = () => <div>{text}</div>;
    const client = await getTestClientWithUser(user);
    const { container } = renderComponent({
      chatCtx: { client },
      messageCtx: { message: deliveredMsg },
      props: { MessageDeliveredStatus },
    });
    expect(container.children[0]).toHaveClass(rootClassName);
    expect(container.children[0]).toBeEmptyDOMElement();
  });

  it('renders default read UI for the last message', async () => {
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: {
        lastReceivedId: deliveredMsg.id,
        message: deliveredMsg,
        readBy: readByOthers,
      },
    });
    expect(screen.getByTestId(MESSAGE_STATUS_READ_TEST_ID)).toMatchSnapshot();
  });
  it('renders custom read UI for the last message', async () => {
    const text = 'CustomMessageReadStatus';
    const MessageReadStatus = () => <div>{text}</div>;
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: {
        lastReceivedId: deliveredMsg.id,
        message: deliveredMsg,
        readBy: readByOthers,
      },
      props: { MessageReadStatus },
    });
    expect(screen.getByText(text)).toBeInTheDocument();
  });
  it('renders empty container without default read UI for not the last message', async () => {
    const client = await getTestClientWithUser(user);
    const { container } = renderComponent({
      chatCtx: { client },
      messageCtx: { message: deliveredMsg },
    });
    expect(container.children[0]).toHaveClass(rootClassName);
    expect(container.children[0]).toBeEmptyDOMElement();
  });

  it('renders empty container without custom read UI for not the last message', async () => {
    const text = 'CustomMessageReadStatus';
    const MessageReadStatus = () => <div>{text}</div>;
    const client = await getTestClientWithUser(user);
    const { container } = renderComponent({
      chatCtx: { client },
      messageCtx: { message: deliveredMsg },
      props: { MessageReadStatus },
    });
    expect(container.children[0]).toHaveClass(rootClassName);
    expect(container.children[0]).toBeEmptyDOMElement();
  });

  it('renders custom Avatar in default read status', async () => {
    const text = 'CustomAvatar';
    const Avatar = () => <div>{text}</div>;
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: {
        lastReceivedId: deliveredMsg.id,
        message: deliveredMsg,
        readBy: readByOthers,
      },
      props: { Avatar },
    });
    expect(screen.getByText(text)).toBeInTheDocument();
    expect(screen.queryByTestId('avatar')).not.toBeInTheDocument();
  });

  it('reflects messageType prop in root class', async () => {
    const client = await getTestClientWithUser(user);
    const { container } = renderComponent({
      chatCtx: { client },
      messageCtx: { message: deliveredMsg },
      props: { messageType: 'XXX' },
    });
    expect(container.children[0]).not.toHaveClass('str-chat__message-simple-status');
    expect(container.children[0]).toHaveClass('str-chat__message-XXX-status');
  });

  it('does not render count if read by own user only', async () => {
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { message: ownMessage, readBy: [user] },
    });
    expect(screen.queryByTestId(MESSAGE_STATUS_READ_COUNT_TEST_ID)).not.toBeInTheDocument();
  });

  it('does not render count if read by 1 other user', async () => {
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { message: ownMessage, readBy: [generateUser()] },
    });
    expect(screen.queryByTestId(MESSAGE_STATUS_READ_COUNT_TEST_ID)).not.toBeInTheDocument();
  });

  it('renders count if read by 2 other users', async () => {
    const client = await getTestClientWithUser(user);
    renderComponent({
      chatCtx: { client },
      messageCtx: { message: ownMessage, readBy: [generateUser(), generateUser()] },
    });
    expect(screen.getByTestId(MESSAGE_STATUS_READ_COUNT_TEST_ID)).toHaveTextContent('2');
  });
});
