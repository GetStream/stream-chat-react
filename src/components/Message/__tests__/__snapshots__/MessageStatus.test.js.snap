// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageStatus theme v1 renders default delivered UI for the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-received"
>
  <div
    class="str-chat__tooltip"
  >
    Delivered
  </div>
  <svg
    height="16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zm3.72 6.633a.955.955 0 1 0-1.352-1.352L6.986 8.663 5.633 7.31A.956.956 0 1 0 4.28 8.663l2.029 2.028a.956.956 0 0 0 1.353 0l4.058-4.058z"
      fill="#006CFF"
      fill-rule="evenodd"
    />
  </svg>
</span>
`;

exports[`MessageStatus theme v1 renders default read UI for the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-read-by"
>
  <div
    class="str-chat__tooltip"
  >
    other-user 
  </div>
  <div
    class="str-chat__avatar str-chat__avatar--circle str-chat__message-sender-avatar"
    data-testid="avatar"
    style="flex-basis: 15px; font-size: 7.5px; height: 15px; line-height: 15px; width: 15px;"
    title="other-user"
  >
    <div
      class="str-chat__avatar-fallback"
      data-testid="avatar-fallback"
    >
      o
    </div>
  </div>
</span>
`;

exports[`MessageStatus theme v1 renders default sending UI for not the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-sending"
>
  <div
    class="str-chat__tooltip"
  >
    Sending...
  </div>
  <div
    class="str-chat__loading-indicator"
    data-testid="loading-indicator-wrapper"
    style="height: 15px; width: 15px;"
  >
    <svg
      height="15"
      viewBox="0 0 30 30"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <lineargradient
          id="a"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stop-color="#FFF"
            stop-opacity="0"
          />
          <stop
            data-testid="loading-indicator-circle"
            offset="100%"
            stop-color="#006CFF"
            stop-opacity="1"
            style="stop-color: rgb(0, 108, 255);"
          />
        </lineargradient>
      </defs>
      <path
        d="M2.518 23.321l1.664-1.11A12.988 12.988 0 0 0 15 28c7.18 0 13-5.82 13-13S22.18 2 15 2V0c8.284 0 15 6.716 15 15 0 8.284-6.716 15-15 15-5.206 0-9.792-2.652-12.482-6.679z"
        fill="url(#a)"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</span>
`;

exports[`MessageStatus theme v1 renders default sending UI for the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-sending"
>
  <div
    class="str-chat__tooltip"
  >
    Sending...
  </div>
  <div
    class="str-chat__loading-indicator"
    data-testid="loading-indicator-wrapper"
    style="height: 15px; width: 15px;"
  >
    <svg
      height="15"
      viewBox="0 0 30 30"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <lineargradient
          id="a"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stop-color="#FFF"
            stop-opacity="0"
          />
          <stop
            data-testid="loading-indicator-circle"
            offset="100%"
            stop-color="#006CFF"
            stop-opacity="1"
            style="stop-color: rgb(0, 108, 255);"
          />
        </lineargradient>
      </defs>
      <path
        d="M2.518 23.321l1.664-1.11A12.988 12.988 0 0 0 15 28c7.18 0 13-5.82 13-13S22.18 2 15 2V0c8.284 0 15 6.716 15 15 0 8.284-6.716 15-15 15-5.206 0-9.792-2.652-12.482-6.679z"
        fill="url(#a)"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</span>
`;

exports[`MessageStatus theme v2 renders default delivered UI for the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-received"
>
  <svg
    data-testid="delivered-icon"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M8.9999 16.2L4.7999 12L3.3999 13.4L8.9999 19L20.9999 6.99998L19.5999 5.59998L8.9999 16.2Z"
      fill="black"
      fill-rule="evenodd"
    />
  </svg>
</span>
`;

exports[`MessageStatus theme v2 renders default read UI for the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-read-by"
>
  <div
    class="str-chat__avatar str-chat__avatar--circle str-chat__message-sender-avatar"
    data-testid="avatar"
    style="flex-basis: 15px; font-size: 7.5px; height: 15px; line-height: 15px; width: 15px;"
    title="other-user"
  >
    <div
      class="str-chat__avatar-fallback"
      data-testid="avatar-fallback"
    >
      o
    </div>
  </div>
</span>
`;

exports[`MessageStatus theme v2 renders default sending UI for not the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-sending"
>
  <div
    class="str-chat__loading-indicator"
    data-testid="loading-indicator-wrapper"
    style="height: 15px; width: 15px;"
  >
    <svg
      height="15"
      viewBox="0 0 30 30"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <lineargradient
          id="a"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stop-color="#FFF"
            stop-opacity="0"
          />
          <stop
            data-testid="loading-indicator-circle"
            offset="100%"
            stop-color="#006CFF"
            stop-opacity="1"
            style="stop-color: rgb(0, 108, 255);"
          />
        </lineargradient>
      </defs>
      <path
        d="M2.518 23.321l1.664-1.11A12.988 12.988 0 0 0 15 28c7.18 0 13-5.82 13-13S22.18 2 15 2V0c8.284 0 15 6.716 15 15 0 8.284-6.716 15-15 15-5.206 0-9.792-2.652-12.482-6.679z"
        fill="url(#a)"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</span>
`;

exports[`MessageStatus theme v2 renders default sending UI for the last message 1`] = `
<span
  class="str-chat__message-simple-status str-chat__message-status"
  data-testid="message-status-sending"
>
  <div
    class="str-chat__loading-indicator"
    data-testid="loading-indicator-wrapper"
    style="height: 15px; width: 15px;"
  >
    <svg
      height="15"
      viewBox="0 0 30 30"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <lineargradient
          id="a"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stop-color="#FFF"
            stop-opacity="0"
          />
          <stop
            data-testid="loading-indicator-circle"
            offset="100%"
            stop-color="#006CFF"
            stop-opacity="1"
            style="stop-color: rgb(0, 108, 255);"
          />
        </lineargradient>
      </defs>
      <path
        d="M2.518 23.321l1.664-1.11A12.988 12.988 0 0 0 15 28c7.18 0 13-5.82 13-13S22.18 2 15 2V0c8.284 0 15 6.716 15 15 0 8.284-6.716 15-15 15-5.206 0-9.792-2.652-12.482-6.679z"
        fill="url(#a)"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</span>
`;
